src/main/resources/scripts/announcements/generate_json.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/python
+import argparse
+import json
+import mysql.connector
+import re
+
+# This script will fetch all the rows from the announcement table, then parse
+# the title out from the message content and return a json file with the newly
+# populated title and modified message content.
+#
+# Step #1: Run this script to generate a json file with the modified
+# announcement data.
+#
+# Step #2: Please review the json file to make sure the title was parsed
+# correctly from the message content.
+#
+# Step #3: To update the announcement table with the json file run the
+# update_announcement.py script next.
+
+def create_json_file(json_file, host, user, password, database):
+
+    json_data = []
+    json_file = open(json_file,"w+")
+
+    db = mysql.connector.connect(
+        host=host,
+        user=user,
+        password=password,
+        database=database
+    )
+    cursor = db.cursor()
+    cursor.execute("SELECT id, created_date, created_by_id, title, message, priority FROM announcement")
+    result = cursor.fetchall()
+    cursor.close()
+    db.close()
+
+    for id, created_date, created_by_id, title, message, priority in result:
+        
+        if not title:
+            lines = message.split("\n",1)
+
+            if len(lines) > 0:
+                first_line = lines[0]
+                header = re.match(r"^#+\s(.+)$", first_line)
+                
+                if header:
+                    title = header[1]
+
+                    if len(lines) > 1:
+                        message = lines[1]
+                else:
+                    if len(lines) > 1:
+                        lines = lines[1].split("\n",1)
+
+                        if len(lines) > 0:
+                            second_line = lines[0]
+                            underline = re.match(r"^(-|=)+$", second_line)
+
+                            if underline:
+                                title = first_line
+
+                                if len(lines) > 1:
+                                    message = lines[1]
```

Comment:
Since we've got this funky regex stuff going on in here can you add some inline comments on how this is working?

Reply:
Yeah, lots if if statements. Please see 649d36421ac3535ab3a16283aac93e9e96130a64.